system-test/datastore.ts

```diff
@@ -811,6 +812,83 @@ describe('Datastore', () => {
       const [entities] = await datastore.runQuery(q);
       assert.strictEqual(entities!.length, 6);
     });
+    describe('with the filter function using the Filter class', () => {
+      it('should run a query with one property filter', async () => {
+        const filter = new PropertyFilter('family', '=', 'Stark');
+        const q = datastore
+          .createQuery('Character')
+          .filter(filter)
+          .hasAncestor(ancestor);
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 8);
+      });
+      it('should run a query with two property filters', async () => {
+        const q = datastore
+          .createQuery('Character')
+          .filter(new PropertyFilter('family', '=', 'Stark'))
+          .filter(new PropertyFilter('appearances', '>=', 20));
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 6);
+      });
+      it('should run a query using new Filter class with filter', async () => {
+        const q = datastore
+          .createQuery('Character')
+          .filter('family', 'Stark')
+          .filter(new PropertyFilter('appearances', '>=', 20));
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 6);
+      });
+      it('should run a query using an AND composite filter', async () => {
+        const q = datastore
+          .createQuery('Character')
+          .filter(
+            AND([
+              new PropertyFilter('family', '=', 'Stark'),
+              new PropertyFilter('appearances', '>=', 20),
+            ])
+          );
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 6);
+      });
+      it('should run a query using an OR composite filter', async () => {
+        const q = datastore
+          .createQuery('Character')
+          .filter(
+            OR([
+              new PropertyFilter('family', '=', 'Stark'),
+              new PropertyFilter('appearances', '>=', 20),
+            ])
+          );
+        const [entities] = await datastore.runQuery(q);
+        assert.strictEqual(entities!.length, 8);
```

Comment:
Is it possible to assert that in resulting query, there are entities that have value `family = Stark` and not `appearances >=20`?

Reply:
Right now I am seeing that all the entities which are returned have a family name of Stark.

Reply:
Done in last commit